WcaOnRails/app/models/competition.rb

```diff
@@ -107,7 +108,11 @@ def delegates_must_be_delegates
     end
   end
 
-  def warnings
+  def user_should_post_delegate_report?(user)
+    !new_record? && is_over? && !delegate_report.posted? && delegates.include?(user)
+  end
```

Comment:
Maybe we can use `delegate_report.can_be_posted?` instead of `is_over? && !delegate_report.posted?` (I guess that `!posted?` should go inside the `can_be_posted?` method in the `DelegateReport` model).